cmd/common/transaction.go

```diff
@@ -240,9 +241,18 @@ func SignParaTimeTransaction(
 	PrintTransactionBeforeSigning(npa, tx)
 
 	// Sign the transaction.
-	sigCtx := signature.DeriveChainContext(npa.ParaTime.Namespace(), npa.Network.ChainContext)
+	// If hardware wallet used follow signing procedure as defined in ADR 14.
 	ts := tx.PrepareForSigning()
-	if err := ts.AppendSign(sigCtx, wallet.Signer()); err != nil {
+	var err error
+	if wallet.UnsafeExport() == "" {
```

Comment:
This doesn't seem like the best API as the caller now needs to handle an internal detail of the underlying Ledger protocol (e.g. CBOR serialization of the metadata field).

Could we instead rename `HwSignRtMeta` to something like `ExtendedContext` (given that it lives in the context) and then make the signers accept a new context interface for the ctx, e.g.:
```golang
type SignContext interface {
    // Extended returns the extended signing context if available. Otherwise it returns nil.
    Extended() *ExtendedContext

    // Derive derives the raw signing context.
    Derive() []byte
}
```

This interface would be implemented for both the existing `Context` type (where `Extended()` could just return `nil`) and the new `ExtendedContext` type (where `Extended()` would return itself and `Derive` would use the `DeriveChainContext` method).

Then you could propagate this new interface all the way to the `Signer` and the Ledger implementation can then try calling `.Extended()` to get the extended context and then handle serialization there. Other signers (e.g. file) can use `Derive()` to get the underlying context.